compass_app/app/lib/ui/search_form/widgets/search_form_continent_carousel.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+import '../../../data/models/continent.dart';
+import '../../core/themes/colors.dart';
+import '../../core/themes/dimens.dart';
+import '../view_models/search_form_viewmodel.dart';
+
+class SearchFormContinentCarousel extends StatelessWidget {
+  const SearchFormContinentCarousel({
+    super.key,
+    required this.viewModel,
+  });
+
+  final SearchFormViewModel viewModel;
+
+  @override
+  Widget build(BuildContext context) {
+    return ListenableBuilder(
+      listenable: viewModel,
+      builder: (context, child) {
+        return ListView.separated(
+          scrollDirection: Axis.horizontal,
+          itemCount: viewModel.searchFormState.continents.length,
+          padding: Dimens.of(context).edgeInsetsScreenHorizontal,
+          itemBuilder: (BuildContext context, int index) {
+            final continent = viewModel.searchFormState.continents[index];
+            return _CarouselItem(
+              key: ValueKey(continent.name),
+              continent: continent,
+              isSelectedContinent:
+                  viewModel.searchFormState.selectedContinent == continent,
+              updateSelectedContinentCommand: () =>
+                  viewModel.updateSelectedContinent(continent),
```

Comment:
_CarouselItem widget would need the Continent and the ViewModel passed in, so I just passed in the properties it needs from the VM instead. This is the only place where the ViewModel isn't passed around. There may be a good reason not to do this.